### PR TITLE
Re-organized into more procedural approach and split into multiple files.

### PR DESCRIPTION
**Pull request by nickmessing**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "javascript.validate.enable": true,
+  "editor.formatOnSave": true,
+  "editor.tabSize": 2,
+  "prettier.singleQuote": true,
+  "prettier.trailingComma": "none",
+  "prettier.printWidth": 120,
+  "prettier.semi": false
+}
```
Comment (egoist) [inline .vscode/settings.json:9]: Do we really need this 😅

File: test/test.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { transform } from 'babel-core'
 
 const transpile = input =>
   transform(input, {
-    plugins: ['syntax-class-properties', 'syntax-object-rest-spread', 'syntax-jsx', './index']
+    plugins: ['syntax-class-properties', 'syntax-object-rest-spread', 'syntax-jsx', './dist/bundle-test']
```
Comment (egoist) [inline test/test.js:6]: why `bundle` instead of `src/index.js`?

File: .vscode/settings.json (hunk 1 above)
Comment (nickmessing) [reply .vscode/settings.json:9]: It's recommended by vscode to commit it, we have just prettier rules here which are good to be consistent

File: test/test.js (hunk 2 above)
Comment (nickmessing) [reply test/test.js:6]: Because Ava uses Babel only for test files and because of Istanbul plugin in babelr for test

Comment (egoist) [reply test/test.js:6]: let AVA transform app files as well then?

Comment (nickmessing) [reply test/test.js:6]: No, that's why we use rollup, Babel and Istanbul plugin to build test bundle before feeding it to Ava

File: .vscode/settings.json (hunk 1 above)
Comment (egoist) [reply .vscode/settings.json:9]: so you prefer format on save over eslint-config-prettier? 😃

File: test/test.js (hunk 2 above)
Comment (egoist) [reply test/test.js:6]: hmm, that's fine, but I personally only use Rollup for production/publish and let test runner transform all files on the fly 😃

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 node_modules
 .nyc_output
 coverage
+dist/bundle-test.js
```
Comment (egoist) [inline .gitignore:4]: seems we don't need to commit `dist/bundle.js` as well

File: .vscode/settings.json (hunk 1 above)
Comment (nickmessing) [reply .vscode/settings.json:9]: @egoist, yep, it just feels nice in vscode, and we don't have `.eslintrc`

File: .gitignore (hunk 3 above)
Comment (nickmessing) [reply .gitignore:4]: Seems legit, will remove

File: .vscode/settings.json (hunk 1 above)
Comment (egoist) [reply .vscode/settings.json:9]: because we're using XO you don't need config file 😄

Comment (egoist) [reply .vscode/settings.json:9]: hmm but not everyone uses vscode..

Comment (nickmessing) [reply .vscode/settings.json:9]: Don't you? We're just 2 for now :)

Comment (egoist) [reply .vscode/settings.json:9]: I sometimes use sublime, well this doesn't matter to me though 😄

btw will vscode prettier be conflict with eslint?

**Comment by aminya** [comment]
Late to the party. What was the status of this? I am interested in using this plugin as a starting point for some other babel plugin. This pull request seems more organized than the big file in the master.

@nickmessing

**Pull request closed without merging**